### PR TITLE
Rails: Update link texts to improve accessibility

### PR DESCRIPTION
**Pull request by mathdebate09**
## Because

Revise the link texts as per the instructions given in the issue.

## This PR

- Update link texts to the sections [Websockets and Action Cable](https://www.theodinproject.com/lessons/ruby-on-rails-websockets-and-actioncable) && [Basic Messaging App (actioncable_lesson.md)](https://github.com/TheOdinProject/curriculum/blob/main/ruby_on_rails/mailers_advanced_topics/actioncable_lesson.md) && [Project: Rails Final Project](https://www.theodinproject.com/lessons/ruby-on-rails-rails-final-project) && [Conclusion](https://www.theodinproject.com/lessons/ruby-on-rails-conclusion) as per the given [spreadsheet](https://docs.google.com/spreadsheets/d/1rM9tjPIFZbKoMfd6OlLnjOhddBe2bivllY8_Bg5aGqY/edit?usp=sharing)

## Issue

Related to [#28290](https://github.com/TheOdinProject/curriculum/issues/28290)

## Additional Information

nil

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by mathdebate09** [comment]
Should I fix the linting issues as well? The issue mentioned to let them be

**Comment by MaoShizhong** [comment]
For these, I can trust you to address any reasonable lint errors. Most should be fixable using the appropriate npm fix scripts, then if any obvious ones remain, you can manually fix those.

Feel free to ask any questions if you encounter any less clear lint errors.

**Comment by mathdebate09** [comment]
Yes they were really easy fixes, but the linter still screams, it requires a section on lesson overview, can i work on it as well? Also there's an error which didn't make sense to me

- ruby_on_rails/mailers_advanced_topics/websockets_and_actioncable.md:320 TOP004/lesson-headings Required heading structure [Expected: h4 heading; Actual: h3 heading] https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP004.md

It's a `Knowledge Check` section which is always written with h3, should i change it to h4 as per the linter?

**Comment by mathdebate09** [comment]
I believe this should do it, also I moved the `Conclusion` up, right before `Assignments` section to adhere to the lesson structure.

File: ruby_on_rails/mailers_advanced_topics/websockets_and_actioncable.md [hunk 1]
```diff
@@ -99,22 +99,21 @@ A channel is a logical unit of work, not too different from what a controller do
 
 Each channel you create can be subscribed to by one or more clients. Messages can then be broadcast to that channel who will forward it to any subscribers.
 
-
 ### Client side concerns
 
-#### Connections
+#### Client-Server connections
```
Comment (MaoShizhong) [inline ruby_on_rails/mailers_advanced_topics/websockets_and_actioncable.md:104]: Should be good to merge after this

```suggestion
#### Client-server connections
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]

**Pull request merged**